src/transformStateWithClones.js

```diff
@@ -10,32 +10,30 @@ function transformStateWithClones(state, actions) {
   const stateWithClone = [];
   const copyState = { ...state };
 
-  function clone(obj) {
-    return { ...obj };
-  }
-
-  actions.forEach(action => {
-    switch (action.type) {
+  actions.forEach(({ type, extraData, keysToRemove }) => {
+    switch (type) {
       case 'addProperties':
-        Object.assign(copyState, action.extraData);
-        stateWithClone.push(clone(copyState));
+        Object.assign(copyState, extraData);
+        stateWithClone.push({ ...copyState });
```

Comment:
Use this expression only once after the switch case, you don't need to repeat yourself in each of the cases